src/templates/docPage.js

```diff
@@ -78,6 +84,33 @@ const BasicDoc = ({ data, location, pageContext }) => {
   if (isTutorial) {
     type = 'tutorial';
   }
+  const useBannerDismissed = createLocalStorageStateHook(
+    `docBannerDismissed-${title}`
+  );
+
+  const { loggedIn } = useLoggedIn();
+
+  const [bannerDismissed, setBannerDismissed] = useBannerDismissed(null);
+  const [bannerVisible, setBannerVisible] = useState(bannerDismissed !== true);
```

Comment:
i think we can remove `bannerVisible` from state. since it can't be set independently (it wouldn't make sense for `bannerVisible` to be anything other than false if `bannerDismissed` or `loggedIn` are true), it can just be a derived value

```suggestion
  const bannerVisible = !loggednIn && !bannerDismissed
```

Reply:
hoohooo so true